From 63c5e9f6ce92f789d6b08a70d2f2abdaf9dc3b0c Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 22 Oct 2024 11:01:25 +0100
Subject: [PATCH] build: Add new dependency on json-c

This will eventually replace Jansson for all JSON parsing.  However
this commit simply introduces the new dependency in the configure
script and documents it.

I chose json-c 0.14 as the baseline since that is the version in RHEL 9.
Probably earlier versions would work.

(cherry picked from libguestfs commit 53872a0a1a267040677572c30b68bd1e8b62ebe3)

(cherry picked from commit 0d7fe8a0b0b429152ea02fc7a7e4f5a1b0eba590)
---
 m4/guestfs-libraries.m4 | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/m4/guestfs-libraries.m4 b/m4/guestfs-libraries.m4
index 32f93afda..b76511982 100644
--- a/m4/guestfs-libraries.m4
+++ b/m4/guestfs-libraries.m4
@@ -167,6 +167,9 @@ LIBS="$old_LIBS"
 dnl Check for Jansson JSON library (required).
 PKG_CHECK_MODULES([JANSSON], [jansson >= 2.7])
 
+dnl Check for JSON-C library (required).
+PKG_CHECK_MODULES([JSON_C], [json-c >= 0.14])
+
 dnl Check for libosinfo (mandatory)
 PKG_CHECK_MODULES([LIBOSINFO], [libosinfo-1.0])
 
